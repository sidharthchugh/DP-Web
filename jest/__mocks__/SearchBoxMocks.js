import React from 'react';
import FontAwesome from 'react-fontawesome';
import {Creatable} from 'react-select';
import DropdownList from 'react-widgets/lib/DropdownList';
import Geosuggest from 'react-geosuggest';
import DateTimePicker from 'react-widgets/lib/DateTimePicker';
import NumberPicker from 'react-widgets/lib/NumberPicker';
import Textarea from 'react-textarea-autosize';

const options = [
    { value: 'one', label: 'One' },
    { value: 'two', label: 'Two' }
];

const mockChooseEditType = (value, i) => {
  return (
    <div className="edit-field">
      {(() => {
        if (value.type === 'dropdown') {
          return (
            <DropdownList name={value.name} data={value.typeValues} placeholder={value.label} />
          );
        } else if (value.type === 'dropdownSearch') {
          return (
            <DropdownList name={value.name} data={value.typeValues} caseSensitive={false} minLength={2} filter="contains" placeholder={value.label} />
          );
        } else if (value.type === 'string') {
          return (
            <div>
              <label>{value.label}</label>
              <div>
                <input placeholder={value.label} type={value.type} />
              </div>
            </div>
          );
        } else if (value.type === 'number') {
          return (
            <NumberPicker name={value.name} placeholder={value.label} />
          );
        } else if (value.type === 'calandar') {
          return (
            <DateTimePicker name={value.name} placeholder={value.label} />
          );
        } else if (value.type === 'phoneNumber') {
          return (
            <div>
              <label>{value.label}</label>
              <div>
                <input placeholder={value.label} type={'number'} />
              </div>
            </div>
          );
        } else if (value.type === 'multiSelect') {
         return (
           <Creatable multi value={null} options={options} onChange={() => {}} placeholder={value.label} />
          );
        } else if (value.type === 'textarea') {
          return (
            <Textarea placeholder={value.label} defaultValue={''} onChange={() => {}} minRows={3} maxRows={6} />
          );
        } else if (value.type === '(cities)' || value.type === '(regions)' || value.type === 'address') {
          return (
            <Geosuggest placeholder={value.label} types={[`${value.type}`]} initialValue={''} onSuggestSelect={() => {}} />
          );
        } else if (value.type === 'fieldArrays') {
          const renderEmployee = ({ fields }) => ( // eslint-disable-line
            <ul className="add-field-list">
              {fields.map((member, index) =>
                <div key={index}>
                  <li>
                    <div className="row">
                      <div className="small-12 medium-6 columns">
                        <input
                          type="text"
                          defaultValue={value.fieldName[i]}
                          onBlur={(event) => this._handleChange(`${member.replace('[', '.').replace(']', '.')}name`, event.target.value)}
                           />
                      </div>
                      <div className="small-12 medium-6 columns">
                        <input
                          type="text"
                          defaultValue={value.fieldPosition[i]}
                          onBlur={(event) => this._handleChange(`${member.replace('[', '.').replace(']', '.')}position`, event.target.value)}
                          />
                      </div>
                    </div>
                    <a
                      className="remove-button"
                      type="button"
                      onClick={() => fields.remove(index)}>Remove
                    </a>
                  </li>
                </div>
                )}
              <div className="add-field">
                <FontAwesome name="plus-circle" />
                <a type="button" onClick={() => fields.push({})}>Add</a>
              </div>
            </ul>
          );
          return (
            <div className="add-input">
              <FieldArray name={value.name} component={renderEmployee} />
            </div>
           );
         }
      })()}
    </div>
  );
};

export {mockChooseEditType};
