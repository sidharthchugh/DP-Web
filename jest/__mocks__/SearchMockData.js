const mockSearchDetails = {
  searchTypes: {
    name: 'searchTypeDropdown',
    label: 'Choose your type of search',
    value: null,
    type: 'dropdown',
    typeValues: {
      auto: 'Automatic - Use my profile',
      productAuto: 'Products/Services - Search interesting options for my company',
      productSpecific: 'Products/Services - I want to specify the details of what I am looking for',
      partnerAuto: 'Strategic partners - Search relevant options for my company',
      partnerSpecific: 'Strategic partners - I want to specify the details of what I am looking for',
      otherAuto: 'Other - I am interested in other relevant opportunities',
      otherSpecific: 'Other - I want to specify other search details'
    },
    id: null
  },
  productSearch: {
    productSearchName: {
      name: 'productSearchName',
      label: 'Search Name',
      value: null,
      type: 'string'
    },
    businessObjectives: {
      name: 'businessObjectives',
      label: 'Business Objectives',
      value: null,
      type: 'multiSelect'
    },
    operationalGoal: {
      name: 'operationalGoal',
      label: 'Operational Goal / Problem',
      value: null,
      type: 'textarea'
    },
    budget: {
      name: 'budget',
      label: 'Budget',
      value: null,
      type: 'string'
    },
    timeline: {
      name: 'timeline',
      label: 'Timeline',
      value: null,
      type: 'string'
    },
    digitalizationScope: {
      name: 'digitalizationScope',
      label: 'Digitalization Scope',
      value: null,
      type: 'textarea'
    },
    capacity: {
      name: 'capacity',
      label: 'CAPACITY REQUIREMENTS',
      value: null,
      type: 'string'
    },
    solution: {
      name: 'solution',
      label: 'Solution',
      value: null,
      type: 'multiSelect'
    },
    valueChain: {
      name: 'valueChain',
      label: 'FOR VALUE CHAIN / SUPPORT FUNCTION',
      value: null,
      type: 'dropdown'
    },
    keyActivities: {
      name: 'keyActivities',
      label: 'PERFORMS KEY ACTIVITIES',
      value: null,
      type: 'multiSelect'
    },
    keyResources: {
      name: 'keyResources',
      label: 'IMPROVES KEY RESOURCES',
      value: null,
      type: 'multiSelect'
    },
    technologiesNeeded: {
      name: 'technologiesNeeded',
      label: 'TECHNOLOGIES NEEDED',
      value: null,
      type: 'multiSelect'
    },
    compatibileTechnologies: {
      name: 'compatibileTechnologies',
      label: 'COMPATIBLE TECHNOLOGIES',
      value: null,
      type: 'multiSelect'
    },
    partnerKeyResources: {
      name: 'partnerKeyResources',
      label: 'Company Description',
      value: null,
      type: 'textarea'
    },
    industryStandards: {
      name: 'industryStandards',
      label: 'PRODUCT / INDUSTRY STANDARDS',
      value: null,
      type: 'multiSelect'
    },
    dataSecurity: {
      name: 'dataSecurity',
      label: 'DATA SECURITY CONCERN',
      value: null,
      type: 'textarea'
    },
    legalConsideration: {
      name: 'legalConsideration',
      label: 'LEGAL CONSIDERATIONS',
      value: null,
      type: 'textarea'
    },
    otherPartnerCharacteristics: {
      name: 'otherPartnerCharacteristics',
      label: 'OTHER PRODUCT REQUIREMENTS',
      value: null,
      type: 'textarea'
    }
  },
  partnerSearch: {
    partnerSearchName: {
      name: 'partnerSearchName',
      label: 'Specific Strategic Partner',
      value: null,
      type: 'string'
    },
    otherPartnerCharacteristics: {
      name: 'otherPartnerCharacteristics',
      label: 'Other Partner Characteristics',
      value: null,
      type: 'multiSelect'
    },
    partnerLanguages: {
      name: 'partnerLanguages',
      label: 'Partner Languages',
      value: null,
      type: 'multiSelect'
    },
    partnerLocation: {
      name: 'partnerLocation',
      label: 'Partner Location',
      value: null,
      type: 'multiSelect'
    },
    partnerSizeIndication: {
      name: 'partnerSizeIndication',
      label: 'Partner Size Indication',
      value: null,
      type: 'multiSelect'
    },
    partnerKeyResources: {
      name: 'partnerKeyResources',
      label: 'Partner Key Resources',
      value: null,
      type: 'multiSelect'
    },
    partnerKeyActivities: {
      name: 'partnerKeyActivities',
      label: 'Partner Key Activites',
      value: null,
      type: 'multiSelect'
    },
    partnerBusinessType: {
      name: 'partnerBusinessType',
      label: 'Partner Business Type',
      value: null,
      type: 'multiSelect'
    },
    partnerSubSector: {
      name: 'partnerSubSector',
      label: 'Partner Sub-Sector',
      value: null,
      type: 'multiSelect'
    },
    partnerSector: {
      name: 'partnerSector',
      label: 'Partner Sector',
      value: null,
      type: 'multiSelect'
    },
    partnerIndustry: {
      name: 'partnerIndustry',
      label: 'Partner Industry',
      value: null,
      type: 'multiSelect'
    },
    partnerBenefits: {
      name: 'partnerBenefits',
      label: 'Benefits to Partner',
      value: null,
      type: 'textarea'
    },
    collaborationRequirements: {
      name: 'collaborationRequirements',
      label: 'Collaboration Requirements',
      value: null,
      type: 'textarea'
    },
    synergies: {
      name: 'synergies',
      label: 'Intended Synergies',
      value: null,
      type: 'textarea'
    },
    assetsSearch: {
      name: 'assetsSearch',
      label: 'Assets we Search',
      value: null,
      type: 'textarea'
    },
    assetsProvide: {
      name: 'assetsProvide',
      label: 'Assets We Provide',
      value: null,
      type: 'textarea'
    },
    operationalGoal: {
      name: 'operationalGoal',
      label: 'Operational Goal / Problem',
      value: null,
      type: 'textarea'
    },
    businessObjectives: {
      name: 'businessObjectives',
      label: 'Business Objectives',
      value: null,
      type: 'multiSelect'
    }
  },
  otherSearch: {
    otherSearchName: {
      name: 'otherSearchName',
      label: 'Other Search',
      value: null,
      type: 'string'
    }
  },
  searchPreferences: {
    suitableProduct: {
      name: null,
      label: 'I am interested in suitable product/service offers',
      value: null,
      type: 'checkbox',
      id: null
    },
    strategicPartners: {
      name: null,
      label: 'I am interested in strategic partnerships',
      value: null,
      type: 'checkbox',
      id: null
    },
    collaboration: {
      name: null,
      label: 'I am interested in joining projects for collaboration',
      value: null,
      type: 'checkbox',
      id: null
    },
    customer: {
      name: null,
      label: 'I am interested in new customers for my product/services',
      value: null,
      type: 'checkbox',
      id: null
    },
    opportunities: {
      name: null,
      label: 'I am interested in other relevant opportunities',
      value: null,
      type: 'checkbox',
      id: null
    }
  }
};

export {
  mockSearchDetails
};
