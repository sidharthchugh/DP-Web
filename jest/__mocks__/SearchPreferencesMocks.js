export const savedPreferences = {
  suitableProduct: false,
  strategicPartners: false,
  collaboration: false,
  customer: false,
  opportunities: false
};
