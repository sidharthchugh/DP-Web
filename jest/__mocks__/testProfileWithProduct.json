{
    "_id" : ObjectId("5853c3eafa9dcae89aa79652"),
    "companyId" : ObjectId("5853c3eafa9dcae89aa79651"),
    "references" : [
        {
            "_id" : ObjectId("5853fb2b45c69401c29aa080"),
            "description" : "description test value",
            "name" : "Project name test value"
        }
    ],
    "projectDescription" : "",
    "projectName" : "",
    "products" : [
        {
            "_id" : ObjectId("5853c3eafa9dcae89aa79653"),
            "companyId" : ObjectId("5853c3eafa9dcae89aa79652"),
            "otherProductAttributes" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea2"),
                    "value" : "other product attributes test value 1",
                    "label" : "other product attributes test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea1"),
                    "value" : "other product attributes test value 2",
                    "label" : "other product attributes test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "segmentsDescription" : "target segments description test value",
            "businessObjective" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea4"),
                    "value" : "solves business objective test value 1",
                    "label" : "solves business objective test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea3"),
                    "value" : "solves business objective test value 2",
                    "label" : "solves business objective test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "dataSecurity" : "data security textarea test value",
            "industryStandards" : "",
            "implementation" : "implementation test value",
            "geographicalMarkets" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea6"),
                    "value" : "geo markets served test value 1",
                    "label" : "geo markets served test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea5"),
                    "value" : "geo markets served test value 2",
                    "label" : "geo markets served test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "maintenanceDifficulty" : "maintenance difficulty & cost test value",
            "integrationDifficulty" : "integration difficulty & related costs test value",
            "delivery" : "delivery/construction time test value",
            "capacityIndication" : "capacity/scalability test value",
            "pricingIndication" : "pricing indication test value",
            "patents" : "patents test value",
            "requirements" : [],
            "compatibilities" : [],
            "technologies" : [
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea8"),
                    "value" : "tech used in the product test value 1",
                    "label" : "tech used in the product test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857bf6ec184293b98616ea7"),
                    "value" : "tech used in the product test value 2",
                    "label" : "tech used in the product test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "uniqueSellingPoints" : "unique selling points test value",
            "valueProposition" : "value proposition test value",
            "productDigitalization" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e33"),
                    "value" : "digitalization scope test value 1",
                    "label" : "digitalization scope test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e32"),
                    "value" : "digitalization scope test value 2",
                    "label" : "digitalization scope test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "keyResources" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e35"),
                    "value" : "reduces/improves key resources test value 1",
                    "label" : "reduces/improves key resources test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e34"),
                    "value" : "reduces/improves key resources test value 2",
                    "label" : "reduces/improves key resources test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "keyActivities" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e37"),
                    "value" : "performs/supports key activities test value 1",
                    "label" : "performs/supports key activities test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e36"),
                    "value" : "performs/supports key activities test value 2",
                    "label" : "performs/supports key activities test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "businessType" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e39"),
                    "value" : "application for business type test value 1",
                    "label" : "application for business type test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e38"),
                    "value" : "application for business type test value 2",
                    "label" : "application for business type test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "applicationValeChain" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3b"),
                    "value" : "value chain position test value 1",
                    "label" : "value chain position test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3a"),
                    "value" : "value chain position test value 2",
                    "label" : "value chain position test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "applicationSector" : "Energy Equipment & Services",
            "applicationIndustry" : "Commercial & Professional Services",
            "problemSolved" : "problem solved test value",
            "solution" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3e"),
                    "value" : "Enterprise Security",
                    "label" : "Enterprise Security",
                    "className" : ""
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3d"),
                    "value" : "Enterprise Networking",
                    "label" : "Enterprise Networking",
                    "className" : ""
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3c"),
                    "value" : "custom production solution value",
                    "label" : "custom production solution value",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "digitalProductCategory" : [
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e40"),
                    "value" : "product category test value 1",
                    "label" : "product category test value 1",
                    "className" : "Select-create-option-placeholder"
                },
                {
                    "_id" : ObjectId("5857ba3dc184293b98616e3f"),
                    "value" : "product category test value 2",
                    "label" : "product category test value 2",
                    "className" : "Select-create-option-placeholder"
                }
            ],
            "productName" : "Product name test value"
        }
    ],
    "digitalizationLevelSupport" : "Digital communication & documentation",
    "supportFunction" : "Information Technology",
    "digitalizationLevelSC" : "Digital communication & documentation",
    "valueChain" : "Operations",
    "otherOnline" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e31"),
            "name" : "Other contact value 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e30"),
            "name" : "Other contact value 2"
        }
    ],
    "twitter" : "@twitter_testvalue",
    "linkedIn" : "linkedin-testvalue",
    "facebook" : "facebook-testvalue",
    "skype" : "skype.testvalue",
    "email" : "email@testvalue.com",
    "phone" : "123456789",
    "companyWebsite" : "website value",
    "totalFundingObtain" : "999999999",
    "revenue" : "999999999",
    "revenueStreams" : "",
    "channels" : "",
    "customerSegments" : "customer segments text area value",
    "customerRelationships" : "customer relationships text area value",
    "valueProposition" : "value proposition text area value",
    "costStructure" : "Cost structure text area value",
    "keyResources" : "key resources text area value",
    "keyActivities" : "Key activities textarea value",
    "otherBusinessRelationships" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2f"),
            "name" : "Other business relationship name 1",
            "type" : "Other business relationship type 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2e"),
            "name" : "Other business relationship name 2",
            "type" : "Other business relationship type 2"
        }
    ],
    "investors" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2d"),
            "name" : "Investor value 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2c"),
            "name" : "Investor value 2"
        }
    ],
    "strategicPartners" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2b"),
            "name" : "Strategic partner name 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e2a"),
            "name" : "Strategic partner name 2"
        }
    ],
    "suppliers" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e29"),
            "name" : "Supplier name 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e28"),
            "name" : "Supplier name 2"
        }
    ],
    "customers" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e27"),
            "name" : "Customer name value 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e26"),
            "name" : "Customer name value 2"
        }
    ],
    "collaborators" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e25"),
            "name" : "Team member name 1",
            "position" : "Job position 1"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e24"),
            "name" : "Team member name 2",
            "position" : "Job position 2"
        }
    ],
    "othercompanyAttributes" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e23"),
            "value" : "My other company attribute 1",
            "label" : "My other company attribute 1",
            "className" : "Select-create-option-placeholder"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e22"),
            "value" : "My other company attribute 2",
            "label" : "My other company attribute 2",
            "className" : "Select-create-option-placeholder"
        }
    ],
    "vision" : "My vision value",
    "mission" : "My mission value",
    "technologies" : "",
    "processesOperations" : "",
    "ftes" : "6-10",
    "languages" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e21"),
            "value" : "German",
            "label" : "German",
            "className" : ""
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e20"),
            "value" : "Hebrew",
            "label" : "Hebrew",
            "className" : ""
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e1f"),
            "value" : "Hindi",
            "label" : "Hindi",
            "className" : ""
        }
    ],
    "streetAddress" : "Ackerstraße",
    "zipCode" : "10115",
    "homeMarket" : "Europe",
    "locations" : [
        {
            "_id" : ObjectId("5857ba3dc184293b98616e1e"),
            "value" : "Location value 1",
            "label" : "Location value 1",
            "className" : "Select-create-option-placeholder"
        },
        {
            "_id" : ObjectId("5857ba3dc184293b98616e1d"),
            "value" : "Location value 2",
            "label" : "Location value 2",
            "className" : "Select-create-option-placeholder"
        }
    ],
    "country" : "Germany",
    "headquarters" : "Berlin",
    "yearEstablished" : "2016",
    "subSector" : "Personal Products",
    "sector" : "IT Services",
    "industry" : "Information Technology",
    "legalForm" : "Limited partnership",
    "businessType" : "Software producer",
    "organizationType" : "Small Enterprise",
    "companyDescription" : "My test company description value",
    "slogan" : "My test slogan value",
    "statusLive": "Not Live",
    "companyName" : "Company Name test",
    "logoURI" : "companylogo-5853c3eafa9dcae89aa79652.png",
    "__v" : 0
}
