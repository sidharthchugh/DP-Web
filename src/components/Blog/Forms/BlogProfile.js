import React, { Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {cmsDetail} from 'actions/cms';
import difference from 'rus-diff';
import BlogLatestProfiles from './BlogLatestProfiles';
import strings from 'components/util/language';
import { Grid } from 'semantic-ui-react';
import Scrollchor from 'react-scrollchor';

class BlogProfile extends Component {
  constructor(props) {
    super(props);
      this.state = { registered: false};
  }

  componentWillMount() {
       this.props.cmsDetail();
  }


  render() {
    const {toggleProfileEdit, fullprofile, cmsData, userLanguage, strings, matchLogo, profileSignUp} = this.props;
    let cmsValue = [];
    if (cmsData.cmsData && cmsData.cmsData.blogs) {
      cmsValue = cmsData.cmsData.blogs.map((feedblog) => {
        feedblog.feedProjectUpatedAt = Date.parse(feedblog.blogDate.replace(/([+\-]\d\d)(\d\d)$/, '$1:$2'));
        feedblog.posted = 'blogfeeds';
        return feedblog;
      });
    }
    const feedArray = [...cmsValue];
    let sortedProfiles = [];
    if (feedArray && feedArray.length > 0) {
      sortedProfiles = feedArray && feedArray.length > 0 && feedArray.sort((a, b) => {
          return b.feedProjectUpatedAt - a.feedProjectUpatedAt;
        });
    }

    return (
      <Grid centered>
        <Grid.Column mobile={12} tablet={12} computer={12}>
      {sortedProfiles.length > 0 && sortedProfiles.map((profile, index) =>
          <BlogLatestProfiles
             toggleProfileEdit={toggleProfileEdit}
             onSubmit={this.saveProfile}
             saveProfile={this.saveProfile}
             profileDetails={fullprofile || profile}
             userLanguage={userLanguage}
             strings={strings}
             cmsData={cmsData}
             profileSignUp={profileSignUp}
             fullprofile
           />
           )}
        </Grid.Column>
      </Grid>
     );
  }
}


// ProfileCard.propTypes = {
//   updateProfiles: PropTypes.func.isRequired,
// };

// Function passed in to `connect` to subscribe to Redux store updates.
// Any time it updates, mapStateToProps is called.
function mapStateToProps(state) {
  return {
     cmsData: state.cms.cmsData
  };
}

// Read more about where to place `connect` here:
// https://github.com/rackt/react-redux/issues/75#issuecomment-135436563
export default connect(mapStateToProps, {cmsDetail})(BlogProfile);
