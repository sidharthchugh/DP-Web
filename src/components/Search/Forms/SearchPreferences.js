import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {updateSearchPreferences} from '../../../actions/searches';
import editIcon from '../../../images/edit-icon.png';
import strings from '../../util/language';
import {Grid, Button} from 'semantic-ui-react';

export class SearchPreferencesComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      disabled: true // Edit mode off by default
    };
    this.toggleEdit = this.toggleEdit.bind(this);
    this.savePreferences = this.savePreferences.bind(this);
    this.renderPreferences = this.renderPreferences.bind(this);
  }


  /**
   * toggleEdit - Enabled/disabled toggle for the checkboxes.
   */
  toggleEdit() {
    this.setState((prevState) => {
      return { disabled: !prevState.disabled };
    });
  }


  /**
   * savePreferences - Converts the form values to an array of
   * objects and dispatches it to the DB.
   *
   * @param  {type} e event object
   */
  savePreferences(e) {
    e.preventDefault();
    // convert HTMLCollection -> Array
    const elements = [].slice.call(this.form.elements);
    // filter out fields with id -> map {id: checked}
    // into a `prefs` object
    let prefs = {};
    elements.filter(elem => elem.id)
            .map((elem) => {
              prefs = { ...prefs, [elem.id]: elem.checked };
              return null;
            });
    // console.log('New passive preference values: ', prefs);
    this.props.updateSearchPreferences(prefs);
    this.setState({ disabled: true });
  }

  renderPreferences() {
    const {searchDetails, savedPreferences, userLanguage} = this.props;
    const {disabled} = this.state;
    const prefKeys = Object.keys(searchDetails.searchPreferences);

    // if we don't have savedPrefs from DB yet -> render nothing
    if (!savedPreferences) return null;

    return prefKeys.map((key) => {
      const item = searchDetails.searchPreferences[key];
      return (
        <Grid.Column
          key={key}
          computer={12}
        >
          <input
            id={key}
            type="checkbox"
            disabled={disabled}
            defaultChecked={savedPreferences[key]}
            className="search-preference-checkbox"
          />
          <label className="search-preferences" htmlFor={key}>{userLanguage === 'German' ? item.germanLabel : item.englishLabel}</label>
        </Grid.Column>
      );
    });
  }


  render() {
    const {disabled} = this.state;
    const {userLanguage} = this.props;
    return (
      <Grid>
        <Grid.Column computer={12} className="search-card search-card--preferences lastCard-generic" id={'search-preferences'}>
          <form
            ref={(c) => { this.form = c; }}
            className={'passive-search-preference' + (disabled ? '' : ' edit-mode')}
          >
            <a className="edit-button edit-button-move" onClick={this.toggleEdit}>
              <img className="edit-icon " src={editIcon} alt="edit icon" /> {strings.edit}
            </a>
            <div className="save-cancel-button">
              <Button type="button" className="button-small button-small--grey" onClick={this.toggleEdit}>{strings.cancel}</Button>
              <Button type="button" className="button-small" type="submit" onClick={this.savePreferences}>{strings.save}</Button>
            </div>

            <div className="card-header">
              <div>
                <h3>{userLanguage === 'German' ? 'Such-Pr√§ferenzen' : 'Search Preferences'}</h3>
              </div>
            </div>
            <Grid className="card-content">
              {this.renderPreferences()}
            </Grid>
          </form>
        </Grid.Column>
      </Grid>
    );
  }
}
SearchPreferencesComponent.propTypes = {
  searchDetails: PropTypes.object.isRequired,
  savedPreferences: PropTypes.object.isRequired
};


// ##############
// REDUX BINDINGS
// ##############
const mapStateToProps = (state) => {
  const {searchPreferences} = state.search.savedSearches;
  return {
    // TODO Ben: we should change the schema to just an Object so we don't
    // need to unnecessarily unbox from an array (which only ever has index 0 anyways).
    savedPreferences: searchPreferences[searchPreferences.length - 1] || {}
  };
};

const SearchPreferences = connect(mapStateToProps, {updateSearchPreferences})(SearchPreferencesComponent);
export default SearchPreferences;
