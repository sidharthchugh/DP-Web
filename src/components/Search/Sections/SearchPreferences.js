const SearchPreferences = {
  searchPreferences: {
    suitableProduct: {
      name: 'suitableProduct',
      englishLabel: '   I am interested in suitable product/service offers',
      germanLabel: '   Ich bin an passenden Produkt/Dienstleistungsangeboten interessiert',
      value: null,
      type: 'checkbox',
    },
    strategicPartners: {
      name: 'strategicPartners',
      englishLabel: '   I am interested in partnerships',
      germanLabel: '   Ich bin an strategischen Partnerschaften interessiert',
      value: null,
      type: 'checkbox',
    },
    collaboration: {
      name: 'collaboration',
      englishLabel: '   I am interested in project-based collaboration',
      germanLabel: '   Ich bin an Projekt-Kollaboration interessiert',
      value: null,
      type: 'checkbox',
    },
    customer: {
      name: 'customer',
      englishLabel: '   I am interested in new customers for my product/services',
      germanLabel: '   Ich bin an neuen Kunden für meine Produkte/Dienstleistungen interessiert',
      value: null,
      type: 'checkbox',
    },
    opportunities: {
      name: 'opportunities',
      englishLabel: '   I am interested in other relevant opportunities',
      germanLabel: '   Ich habe Interesse an anderen, relevanten Geschäftsmöglichkeiten',
      value: null,
      type: 'checkbox',
    }
  }
};

export default SearchPreferences;
