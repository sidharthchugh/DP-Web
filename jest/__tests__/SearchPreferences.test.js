import React from 'react';
import renderer from 'react-test-renderer';
import {SearchPreferencesComponent} from '../../src/components/Search/Forms/SearchPreferences';
import {mockSearchDetails} from '../__mocks__/SearchMockData';
import {savedPreferences} from '../__mocks__/SearchPreferencesMocks';

describe('SearchPreferences', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SearchPreferencesComponent
        searchDetails={mockSearchDetails}
        savedPreferences={savedPreferences}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

  /* it('toggles the edit state between enabled/disabled', () => {
    expect(searchPrefs.state.disabled).toBe(true); // initialState
    searchPrefs.toggleEdit(); // trigger the local state change
    expect(searchPrefs.state.disabled).toBe(false);
  }); */

  /* it('saves changed preferences correctly', () => {
    const e = {preventDefault: () => {}}; // mock the event object
    searchPrefs.form = {};
    expect(searchPrefs.savePreferences(e)).toBeDefined();
  });
 */
