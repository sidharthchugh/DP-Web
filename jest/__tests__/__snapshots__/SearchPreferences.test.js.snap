exports[`SearchPreferences renders correctly 1`] = `
<form
  className="small-12 columns no-padding-left-right search-card no-padding passive-search-preference">
  <a
    className="edit-button"
    onClick={[Function]}>
    <img
      alt="edit-icon"
      className="edit-icon"
      src="test-file-stub" />
     Edit
  </a>
  <div
    className="save-cancel-button">
    <a
      className="hollow small secondary button"
      onClick={[Function]}>
      Cancel
    </a>
    <button
      className="small alert button"
      onClick={[Function]}
      type="submit">
      Save
    </button>
  </div>
  <div
    className="row card-header">
    <div
      className="small-12 columns">
      <h3>
        Passive Search Preferences
      </h3>
    </div>
  </div>
  <div
    className="row card-content">
    <div
      className="small-12 columns checkbox-list-item">
      <input
        defaultChecked={false}
        disabled={true}
        id="suitableProduct"
        type="checkbox" />
      <label
        htmlFor="suitableProduct">
        I am interested in suitable product/service offers
      </label>
    </div>
    <div
      className="small-12 columns checkbox-list-item">
      <input
        defaultChecked={false}
        disabled={true}
        id="strategicPartners"
        type="checkbox" />
      <label
        htmlFor="strategicPartners">
        I am interested in strategic partnerships
      </label>
    </div>
    <div
      className="small-12 columns checkbox-list-item">
      <input
        defaultChecked={false}
        disabled={true}
        id="collaboration"
        type="checkbox" />
      <label
        htmlFor="collaboration">
        I am interested in joining projects for collaboration
      </label>
    </div>
    <div
      className="small-12 columns checkbox-list-item">
      <input
        defaultChecked={false}
        disabled={true}
        id="customer"
        type="checkbox" />
      <label
        htmlFor="customer">
        I am interested in new customers for my product/services
      </label>
    </div>
    <div
      className="small-12 columns checkbox-list-item">
      <input
        defaultChecked={false}
        disabled={true}
        id="opportunities"
        type="checkbox" />
      <label
        htmlFor="opportunities">
        I am interested in other relevant opportunities
      </label>
    </div>
  </div>
</form>
`;
